modules/extensions/src/mask/mask.js

```diff
@@ -20,17 +21,19 @@ export default class MaskExtension extends LayerExtension {
     };
   }
 
-  draw({uniforms, moduleParameters}) {
+  draw({uniforms, context, moduleParameters}) {
     uniforms.mask_maskByInstance = this.state.maskByInstance;
-    const {maskId} = this.props;
+    const {maskId} = this.root.props;
```

Comment:
maskId should be automatically passed down via `LayerExtension.getSubLayerProps`.